CMakeLists.txt

```diff
@@ -77,9 +77,13 @@ if("sgx" IN_LIST COMPILE_TARGETS)
 
   target_include_directories(
     ccf.enclave SYSTEM
-    PUBLIC $<BUILD_INTERFACE:${CCF_GENERATED_DIR}>
-           $<INSTALL_INTERFACE:include/ccf/>
-           $<INSTALL_INTERFACE:include/3rdparty/>
+    PUBLIC
+      $<BUILD_INTERFACE:${CCF_GENERATED_DIR}>
+      $<INSTALL_INTERFACE:include/ccf/> #< This contains the private headers
+                                        #< which are currently under src, and
+                                        #< should be removed or renamed
+      $<INSTALL_INTERFACE:include/>
+      $<INSTALL_INTERFACE:include/3rdparty/>
```

Comment:
I think we would ideally split out the test (doctest, trompeloeil, picobench) and even the host-only 3party headers (CLI11) so they aren't installed.

Reply:
Yes. I've added this as a TODO on #2328. It looks like CMake doesn't make this easy - the `REGEX`/`PATTERN` always installs the directories, even if they're empty, so this becomes quite a verbose description opting in every library we want to keep. There's also a few libraries that we _could_ hide as an implementation detail (`small_vector`, `sss`, `snmalloc`?) once our public includes are clean, but I won't try to unpick that right now.

Reply:
I would have thought we just create sub-directories like '3rdparty/export' and '3rdparty/internal'? I agree this isn't urgent by any means.